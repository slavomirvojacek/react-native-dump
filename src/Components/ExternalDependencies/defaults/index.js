import { noop } from '@openmaths/utils'

const props = {
  logIntoConsole: noop,
}

export {
  props
}
