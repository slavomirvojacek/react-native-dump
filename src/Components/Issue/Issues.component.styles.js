import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  issueLink: {
    color: 'blue',
    fontSize: 16,
    textDecorationLine: 'underline',
    textDecorationStyle: 'solid',
    textDecorationColor: 'blue',
    marginTop: 5,
    marginBottom: 5
  }
})
