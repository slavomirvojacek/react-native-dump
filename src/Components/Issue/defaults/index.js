const issueDefault = (override = {}) => {
  const def = {
    id: '',
    title: '',
    url: '',
    bodyText: '',
    reactions: []
  }

  return {...def, ...override}
}

const props = {
  issue: issueDefault(),
}

export {
  issueDefault, props
}
