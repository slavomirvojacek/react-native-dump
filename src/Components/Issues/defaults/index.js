const props = {
  issues: [],
}

export {
  props
}
