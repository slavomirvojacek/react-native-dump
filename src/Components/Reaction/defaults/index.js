const userDefault = (id = '') => ({id})

const reactionDefault = (override = {}) => {
  const def = {
    id: '',
    content: '',
    user: userDefault()
  }

  return {...def, ...override}
}

const props = {
}

export {
  reactionDefault, props
}
