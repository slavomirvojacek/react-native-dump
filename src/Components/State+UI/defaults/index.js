const props = {
  color: 'black', // has to be a valid colour, otherwise warnings are thrown
  isWarningVisible: false,
}

const state = {
  ...props,
  color: 'green', // has to be a valid colour, otherwise warnings are thrown
}


export {
  props, state
}
