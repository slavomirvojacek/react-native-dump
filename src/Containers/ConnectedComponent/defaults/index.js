import { None } from '@threestup/monads'

const props = {
  showLoading: false,
  username: None,
  error: None,

}

export {
  props
}
