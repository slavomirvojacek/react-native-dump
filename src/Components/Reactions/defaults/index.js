import { None } from '@threestup/monads'

import { issueDefault } from '../../Issue/defaults'

const props = {
  issue: issueDefault(),
  me: None
}

export {
  props
}
