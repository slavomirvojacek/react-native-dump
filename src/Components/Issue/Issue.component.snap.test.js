import Issue from './Issue.component'

import { issueDefault, props as testDefaultProps } from './defaults'
import { assertSnapshots } from '../../TestUtils/snapshot'

jest
  .mock('../Reactions/Reactions.component', () => 'Reactions')
  .mock('../../Utils/misc', () => ({
    openUrl: jest.fn()
  }))

describe('Issue Component Snapshots', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  const configs = [
    {
      props: {...testDefaultProps},
      desc: 'renders correctly'
    },
    {
      props: {
        ...testDefaultProps,
        issue: issueDefault({title: 'Some Title', url: 'https://some.url'}),
      },
      desc: 'renders correctly with specific title and url'
    },
  ]

  assertSnapshots(Issue, configs)
})
