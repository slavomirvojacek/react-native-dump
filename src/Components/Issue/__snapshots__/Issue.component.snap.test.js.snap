// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue Component Snapshots renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "blue",
        "fontSize": 16,
        "marginBottom": 5,
        "marginTop": 5,
        "textDecorationColor": "blue",
        "textDecorationLine": "underline",
        "textDecorationStyle": "solid",
      }
    }
  >
    
  </Text>
  <Reactions
    issue={
      Object {
        "bodyText": "",
        "id": "",
        "reactions": Array [],
        "title": "",
        "url": "",
      }
    }
    reactions={_None {}}
  />
</View>
`;

exports[`Issue Component Snapshots renders correctly with specific title and url 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "blue",
        "fontSize": 16,
        "marginBottom": 5,
        "marginTop": 5,
        "textDecorationColor": "blue",
        "textDecorationLine": "underline",
        "textDecorationStyle": "solid",
      }
    }
  >
    Some Title
  </Text>
  <Reactions
    issue={
      Object {
        "bodyText": "",
        "id": "",
        "reactions": Array [],
        "title": "Some Title",
        "url": "https://some.url",
      }
    }
    reactions={_None {}}
  />
</View>
`;
